### DISCUSSION TITLE
how to overwrite MCP tool error response ?

### DISCUSSION THREAD

**Discussion by lnvestor** [modelcontextprotocol/.github#122] [Q&A]
### Pre-submission Checklist

- [X] I have checked that this question would not be more appropriate as an issue in a specific repository
- [X] I have searched existing discussions and documentation for answers

### Question Category

- [X] Protocol Specification
- [X] SDK Usage
- [X] Server Implementation
- [X] General Implementation
- [ ] Documentation
- [ ] Other

### Your Question

Hi everyone 

i have one tool that i uses API to create 

if the creating is succesfful i see that on the tool but if there is an error i get only this 

{
  `type`: `salesforce`,
  `name`: `Salesforce Production`,
  `salesforce`: {
    `oauth2FlowType`: `authorization_code`,
    `concurrencyLevel`: 5,
    `sandbox`: false,
    `scope`: [
      `api`,
      `refresh_token`
    ]
  },
  `microServices`: {
    `disableDataWarehouse`: false,
    `disableNetSuiteWebServices`: true,
    `disableRdbms`: true
  }
}

**[{"type": "text", "text": "Error executing code: MCP error -32603: MCP error -32603: Celigo API error: Request failed with status code 422"}]**

when i test this in postman here is the error response 

{
    "errors": [
        {
            "field": "filter.expression.version",
            "code": "enum",
            "message": "`1.0.0` is not a valid enum value for path `filter.expression.version`."
        }
    ]
}

Thanks

**Comment by jspahrsummers** [comment]
Errors that occur during tool execution should be reported as tool results, rather than protocol errors: https://spec.modelcontextprotocol.io/specification/server/tools/#error-handling
